src/common.speech/SynthesisAdapterBase.ts

```diff
@@ -0,0 +1,630 @@
+import {ISynthesisConnectionFactory} from "./ISynthesisConnectionFactory";
+import {SpeechConnectionMessage} from "./SpeechConnectionMessage.Internal";
+import {SynthesizerConfig} from "./SynthesizerConfig";
+
+export class SynthesisAdapterBase implements IDisposable {
+    protected privRequestSession: RequestSession;
+    protected privSynthesisTurn: SynthesisTurn;
+    protected privConnectionId: string;
+    protected privSynthesizerConfig: SynthesizerConfig;
+    protected privSpeechSynthesizer: SpeechSynthesizer;
+    protected privSuccessCallback: (e: SpeechSynthesisResult) => void;
+    protected privErrorCallback: (e: string) => void;
+
+    public get synthesisContext(): SpeechContext {
+        return this.privSpeechContext;
+    }
+
+    public get dynamicGrammar(): DynamicGrammarBuilder {
+        return this.privDynamicGrammar;
+    }
+
+    public get agentConfig(): AgentConfig {
+        return this.privAgentConfig;
+    }
+
+    public get connectionEvents(): EventSource<ConnectionEvent> {
+        return this.privConnectionEvents;
+    }
+
+    public get serviceEvents(): EventSource<ServiceEvent> {
+        return this.privServiceEvents;
+    }
+
+    protected speakOverride: (ssml: string, requestId: string, sc: (e: SpeechSynthesisResult) => void, ec: (e: string) => void) => any = undefined;
+
+    // Called when telemetry data is sent to the service.
+    // Used for testing Telemetry capture.
+    public static telemetryData: (json: string) => void;
+    public static telemetryDataEnabled: boolean = true;
+
+    public set activityTemplate(messagePayload: string) { this.privActivityTemplate = messagePayload; }
+    public get activityTemplate(): string { return this.privActivityTemplate; }
+
+    protected receiveMessageOverride: () => any = undefined;
+
+    protected connectImplOverride: (isUnAuthorized: boolean) => any = undefined;
+
+    protected configConnectionOverride: () => any = undefined;
+
+    protected fetchConnectionOverride: () => any = undefined;
+
+    public set audioOutputFormat(format: AudioOutputFormatImpl) {
+        this.privAudioOutputFormat = format;
+        this.privSynthesisTurn.audioOutputFormat = format;
+        if (this.privSessionAudioDestination !== undefined) {
+            this.privSessionAudioDestination.format = format;
+        }
+    }
+    private privAuthentication: IAuthentication;
+    private privConnectionFactory: ISynthesisConnectionFactory;
+
+    // A promise for a configured connection.
+    // Do not consume directly, call fetchConnection instead.
+    private privConnectionConfigurationPromise: Promise<IConnection>;
+
+    // A promise for a connection, but one that has not had the speech context sent yet.
+    // Do not consume directly, call fetchConnection instead.
+    private privConnectionPromise: Promise<IConnection>;
+    private privAuthFetchEventId: string;
+    private privIsDisposed: boolean;
+    private privMustReportEndOfStream: boolean;
```

Comment:
>privMustReportEndOfStream [](start = 12, length = 25)

Is this ever read?

Reply:
ah, I copyed this from `ServiceRecognizerBase` and forget to clean it. Thanks